src/ai/backend/common/events.py

```diff
@@ -1358,132 +1335,101 @@ async def dispatch_subscribers(
     ) -> None:
         if self._log_events:
             log.debug("DISPATCH_SUBSCRIBERS(ev:{}, ag:{})", event_name, source)
-        for subscriber in self.subscribers[event_name].copy():
-            self.subscriber_taskgroup.create_task(
+        for subscriber in self._subscribers[event_name].copy():
+            self._subscriber_taskgroup.create_task(
                 self.handle("SUBSCRIBER", subscriber, source, args),
             )
             await asyncio.sleep(0)
 
     @preserve_termination_log
     async def _consume_loop(self) -> None:
-        async with aclosing(
-            redis_helper.read_stream_by_group(
-                self.redis_client,
-                self._stream_key,
-                self._consumer_group,
-                self._consumer_name,
-            )
-        ) as agen:
-            async for msg_id, msg_data in agen:
-                if self._closed:
-                    return
-                if msg_data is None:
-                    continue
-                event_type = "unknown"
-                start = time.perf_counter()
-                try:
-                    decoded = msg_data[b"name"].decode()
-                    if decoded and isinstance(decoded, str):
-                        event_type = decoded
-                    await self.dispatch_consumers(
-                        decoded,
-                        msg_data[b"source"].decode(),
-                        msgpack.unpackb(msg_data[b"args"]),
-                    )
-                    self._metric_observer.observe_event_success(
-                        event_type=event_type,
-                        duration=time.perf_counter() - start,
-                    )
-                except Exception as e:
-                    self._metric_observer.observe_event_failure(
-                        event_type=event_type,
-                        duration=time.perf_counter() - start,
-                        exception=e,
-                    )
-                    log.exception("EventDispatcher.consume(): unexpected-error")
-                except BaseException as e:
-                    self._metric_observer.observe_event_failure(
-                        event_type=event_type,
-                        duration=time.perf_counter() - start,
-                        exception=e,
-                    )
-                    raise
+        async for msg in self._msg_queue.consume_queue():  # type: ignore
```

Comment:
I think it would be good to add a type hint to the `msg` variable.

```suggestion
        async for msg in self._msg_queue.consume_queue():  # type: ignore
            msg = cast(MQMessage, msg)
```

Reply:
Should we be doing unnecessary casts because of mypy bug?